src/view/menu.js

```diff
@@ -1,21 +1,39 @@
-export const createMenuTemplate = (filmsContainer = {}) => {
-  let watchListLength = 0;
-  let watchedLength = 0;
-  let favoriteLength = 0;
-  filmsContainer.forEach(({userInfo}) => {
-    const {isWatchList, isWatched, isFavorite} = userInfo;
-    if(isWatchList){watchListLength++;}
-    if(isWatched){watchedLength++;}
-    if(isFavorite){favoriteLength++;}
-  });
-
+import {createElement} from '../utils';
+const createMenuTemplate = (allFilms = {}) => {
+  const filmsToFilter = {
+    watchlist: allFilms.filter(({userInfo}) => userInfo.isWatchList).length,
+    history: allFilms.filter(({userInfo}) => userInfo.isWatched).length,
+    favorites: allFilms.filter(({userInfo}) => userInfo.isFavorite).length,
+  };
```

Comment:
> Насчет меню (там где сколько любимых, просмотренных и тд), я сделала через фильтр. Только не очень поняла в чем преимущество. Ну или я не правильно сделала. Если я правильно понимаю, раньше я проходила по массиву один раз, а теперь целых три, что вроде как не логично. 

Посмотри критерий Д25, там явно указано использовать итераторы.

Можно сделать в один проход через reduce (код абстрактный, не проверял, принцип понятен должен быть):

```suggestion
  const [watchListCount, watchedCount, favouriteCount] = allFilms.reduce((result, {userInfo}) => {
    return [
      result[0] + userInfo.isWatchList.toNumber(),
      result[1] + userInfo.isWatched.toNumber(),
      result[2] + userInfo.isFavorite.toNumber(),
      ]
  }, [0,0,0])
```